### ISSUE TITLE
[Question] : How to add a default header map to HttpClientBuilder?

### ISSUE THREAD

**Issue by ansrivas** [sagebind/isahc#180]
Hi, thank you for your work in creating this project.

I am trying to inject a default `HeaderMap` in client builder. I saw that there is `SetOpt` implementation as well but no method in `Configurable` to append the headers.

How can I achieve something like this?
```rust
let mut headers = HeaderMap::new();
headers.insert(
  "X-header-something",
  HeaderValue::from_str("my-value").unwrap(),
);

let c = isahc::HttpClientBuilder::new()
    .set_default_headers(headers) // <---- Something like this
    .build()?;  
```

**Comment by sagebind** [comment]
Great question! This isn't currently possible, but its a great idea for a feature! I imagine it would look something like this:

```rust
impl HttpClientBuilder {
    // etc

    pub fn default_header(self, key: impl TryInto<HeaderName>, value: impl TryInto<HeaderValue>) -> Self {}

    pub fn default_headers_mut(&mut self) -> &mut HeaderMap<HeaderValue> {}
}
```

> I saw that there is `SetOpt` implementation as well

That's just used as an internal helper for setting the headers provided in a `Request` given. Implementing default headers for a client would leverage the same code path.

**Comment by sagebind** [comment]
This feature is now available in 0.9.2! :tada: Thanks to @ansrivas for implementing this!
